lib/src/folding/parsers/spaces_foldable_block_parser.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:collection';
+
+import 'package:charcode/ascii.dart';
+import '../foldable_block.dart';
+import '../foldable_block_type.dart';
+
+import 'abstract.dart';
+
+///A parser for foldable blocks from spaces count at start of each line.
+class SpacesFoldableBlockParser extends AbstractFoldableBlockParser {
```

Comment:
Let's call this `IndentationFoldableBlockParser` because tabs are also a valid chars for this.
If you think you can add support for tabs quickly, then add it please. Otherwise please file an issue to support tabs for this parser.